### DISCUSSION TITLE
zshot spaCy ignoring email and phone entities?

### DISCUSSION THREAD

**Discussion by jakesno** [explosion/spaCy#13419] [Help: Model Advice] [answered]
Hi All,

We are using spaCy for entity and keyword extraction and noticed that it generally ignores phone numbers and emails within our text data. I have included a basic, redacted example below where would have expected the phone number to be extracted. 

Example, "Our Air Conditioning Installation service in XXXX offers top-quality solutions to keep your home or office cool and comfortable. Call us ###-###-#### or email at xxxx@xxxx.com"

Anything im missing?

**Comment by svlandeg** [comment, marked as answer]
Hi!

The pretrained NER model from the [English models](https://spacy.io/models/en) doesn't have specific labels for `Email` or `Phone_number`. It does however recognize `Cardinal` and `Ordinal` entities, which might be relevant for your use-case. They might not capture exactly what you want though.

For this type of information extraction, I would recommend using builtin attributes that spaCy defines on the token-level, such as `like_num` and `like_email`. These can then be used to design custom matcher rules, see [here](https://spacy.io/usage/rule-based-matching#adding-patterns-attributes).

So for instance, you can write something like:
```
    text = "Hello Jake is your email still jake@nowhere.ai and your phone number 0342-42-66?"
    nlp = spacy.load("en_core_web_lg")
    doc = nlp(text)
    for ent in doc.ents:
        print(ent.text, ent.label_)
    print()
    for token in doc:
        if token.like_email:
            print(token, "Email")

    matcher = Matcher(nlp.vocab)
    matcher.add("Phone Number", patterns=[[{"LIKE_NUM": True}, {"IS_PUNCT": True}, {"LIKE_NUM": True}, {"IS_PUNCT": True}, {"LIKE_NUM": True}]])

    matches = matcher(doc, as_spans=True)
    for match in matches:
        print(match.text, match.label_)
```
and you'll get

> Jake PERSON
0342-42 CARDINAL
jake@nowhere.ai Email
0342-42-666 Phone Number
